crates/bevy_render/src/view/window/mod.rs

```diff
@@ -477,7 +448,53 @@ pub fn create_surfaces(
                     }
                 }
 
-                SurfaceData { surface, format }
+                let configuration = wgpu::SurfaceConfiguration {
+                    format,
+                    width: window.physical_width,
+                    height: window.physical_height,
+                    usage: TextureUsages::RENDER_ATTACHMENT,
+                    present_mode: match window.present_mode {
+                        PresentMode::Fifo => wgpu::PresentMode::Fifo,
+                        PresentMode::FifoRelaxed => wgpu::PresentMode::FifoRelaxed,
+                        PresentMode::Mailbox => wgpu::PresentMode::Mailbox,
+                        PresentMode::Immediate => wgpu::PresentMode::Immediate,
+                        PresentMode::AutoVsync => wgpu::PresentMode::AutoVsync,
+                        PresentMode::AutoNoVsync => wgpu::PresentMode::AutoNoVsync,
+                    },
+                    // TODO: Expose this as a setting somewhere
+                    // 2 is wgpu's default/what we've been using so far.
+                    // 1 is the minimum, but may cause lower framerates due to the cpu waiting for the gpu to finish
+                    // all work for the previous frame before starting work on the next frame, which then means the gpu
+                    // has to wait for the cpu to finish to start on the next frame.
+                    desired_maximum_frame_latency: 2,
+                    alpha_mode: match window.alpha_mode {
+                        CompositeAlphaMode::Auto => wgpu::CompositeAlphaMode::Auto,
+                        CompositeAlphaMode::Opaque => wgpu::CompositeAlphaMode::Opaque,
+                        CompositeAlphaMode::PreMultiplied => {
+                            wgpu::CompositeAlphaMode::PreMultiplied
+                        }
+                        CompositeAlphaMode::PostMultiplied => {
+                            wgpu::CompositeAlphaMode::PostMultiplied
+                        }
+                        CompositeAlphaMode::Inherit => wgpu::CompositeAlphaMode::Inherit,
+                    },
+                    view_formats: if !format.is_srgb() {
+                        vec![format.add_srgb_suffix()]
+                    } else {
+                        vec![]
+                    },
+                };
+
+                render_device.configure_surface(&surface, &configuration);
+
+                SurfaceData {
+                    surface,
+                    configuration,
+                }
             });
+
+        if window.size_changed || window.present_mode_changed {
+            render_device.configure_surface(&data.surface, &data.configuration);
```

Comment:
This is somehow resulting in Linux not being able to construct the surface at all on resize. I'm noticing that both this _and_ the configure_surface in the render app (the Outdated case) is getting called. `configure_surface` also appears to be a pretty heavy call. Probably not something we should be doing more than once per frame, even if it wasn't causing these problems.

https://github.com/bevyengine/bevy/assets/2694663/41993f0f-8f1c-469a-ace3-5c5d4cc5844f

Reply:
Or rather: it is "succeeding to construct the surface but also fails to render". It doesn't not "fail to construct the surface" as I said.

Reply:
I hope I fixed the resizing issue. at least I fixed an issue, not sure if that was what you're getting

Reply:
> I'm noticing that both this and the configure_surface in the render app (the Outdated case) is getting called.

I didn't manage to get those two cases happens in the same frame, are you confident on that?

Reply:
Ah yeah you're right. It appears to skip a frame before triggering the Outdated branch. Then it continually triggers the Outdated branch every frame.

Your change does fix the "every frame triggers Outdated branch" problem, as well as the visually broken resizing (which are probably the same problem).

Reply:
I still feel like we shouldn't be handling this event twice (both for potential performance reasons, for the  potential problems it might create on different platforms, and for future issues that might arise from doing this in two places).

Can we consider alternatives, such as just adding this parameter to `prepare_windows` to force it onto the main thread on mac/ios?

```rust
    #[cfg(any(target_os = "macos", target_os = "ios"))] _marker: Option<
        NonSend<bevy_core::NonSendMarker>,
    >
```

Reply:
I would rather not put `prepare_windows` back on the main thread as it would then block the main thread and break pipelined rendering when not in the continuous update mode for the event loop

Reading through wgpu, it seems the `Outdated` case can only happen on Vulkan. I'm not sure why one is triggered after changing the first call to `configure_surface`, not sure this was changed by this PR?

Reply:
Hmmm ok in that case I guess we should just move forward with the current fix.